api/soap/mc_enum_api.php

```diff
@@ -290,11 +290,25 @@ function mci_enum_get_array_by_id( $p_enum_id, $p_enum_type, $p_lang ) {
 		$t_result['name'] = MantisEnum::getLocalizedLabel( $t_enum_string_value, $t_enum_localized_value, $p_enum_id );
 	} else {
 		$t_enum_array = MantisEnum::getAssocArrayIndexedByValues( $t_enum_string_value );
+
+		if (!array_key_exists($p_enum_id, $t_enum_array)) {
+				log_event( LOG_WEBSERVICE, "Id '$p_enum_id' does not exist in the '$p_enum_type' enumeration"  );
+				$t_result['name'] = $t_result['label'] = 'NOT FOUND!';
```

Comment:
I'm not overly keen on the hardcoded `NOT FOUND!`. For consistency with standard MantisBT behavior in such case, I believe the webservice should return `@<value>@` [just like Mantis does](https://github.com/mantisbt/mantisbt/blob/release-2.6.0/core/classes/MantisEnum.class.php#L223).

If a text message really has to be displayed, then it should be localized via *get_lang()* call (and in that case, again for consistency, the change should probably take place in *MantisEnum::getLabelForUnknownValue()* rather than here.